### DISCUSSION TITLE
Lichess API

### DISCUSSION THREAD

**Discussion by shuvashish76** [thomas-mauran/chess-tui#95] [Ideas]
Possibilty of playing chess online via lichess API.
https://lichess.org/source#docs-utilities-others

Would be nice to have feature to play on lichess via thier api directly from chess-tui

**Comment by thomas-mauran** [comment]
Hello @shuvashish76, this is a great idea, maybe you can create an issue mentioning this feature ?

**Reply by shuvashish76** [reply]
Lichess has too many features I thought better to create a discussion for ideas. (_GitHub has the option to create issues from discussion page. Ok Done👍)_

Basic features (_not sure how hard is to implement them_) that come to my mind:
- Playing multiple games (need switch/auto_switch option) e.g. correspondence or simiul
- Playing tournament (i.e. immediately switch to next game when paired), possibility of tournament stats while playing
- Accepting challenges
- Seek game or accept challenge from lobby
- Puzzles, streak, storm, racer
- Coordinates practice
- Run Lichess TV after certain period of user inactivity etc...

**Reply by thomas-mauran** [reply]
Oh that's nice I didn't know you could create issues out of discutions

**Reply by thomas-mauran** [reply]
Those are very interesting features, I think we might need to sort them cause some should be done before others. For example being able to play a game could be split in multiple, 1 being able to login to lichess and store it in the chess tui toml config, 2 being able to join and play a game with an id, 3 being able to create a game. 

I will do a list out of the features you mentioned in this thread later. 

Some features might be hard to implement, like the lichess tv for example which I don't know a lot about, while others like the puzzle one could be very trivial and a big value for the game !

**Reply by thomas-mauran** [reply]
I found this for playing the game:

https://lichess.org/api#tag/Games/operation/streamGame

This seems to use a stream with fen strings, if so we already are using a similar concepts for the chess engine games so this is nice meaning it might be kind of similar to implement

**Comment by shuvashish76** [comment]
Info: Lichess API has some restrictions for cheating reasons
- https://lichess.org/@/lichess/blog/welcome-lichess-bots/WvDNticA
- https://lichess.org/@/lichess/blog/welcome-lichess-boards/XlRW5REA

More info on their discord platform https://lichess.org/run/external-link/discord

**Comment by thomas-mauran** [comment]
Hello @shuvashish76 happy to tell you we finally have that lichess update ! You can now connect your account and play against lichess users !

I am working on the clock system to improve the lichess integration with further time control on games

**Reply by shuvashish76** [reply]
Awesome! I've exams so temporarily disabled my account, will check soon :)

**Reply by thomas-mauran** [reply]
Good luck for your exams !

**Reply by shuvashish76** [reply]
Everything as mentioned works as expected :)
